sdks/python/apache_beam/runners/interactive/interactive_environment.py

```diff
@@ -675,3 +683,25 @@ def get_sql_chain(self, pipeline, set_user_pipeline=False):
             pipeline)
       chain.user_pipeline = pipeline
     return chain
+
+  def check_bucket_exists(self, bucket_name):
+    try:
+      from apitools.base.py.exceptions import HttpError
+    except:
+      raise ImportError('Could not import HttpError from apitools.')
+
+    try:
+      storage_client = storage.StorageV1(
+          credentials=auth.get_service_credentials(),
+          get_credentials=False,
+          http=get_new_http(),
+          response_encoding='utf8')
+      request = storage.StorageBucketsGetRequest(bucket=bucket_name)
+      storage_client.buckets.Get(request)
+    except HttpError as e:
+      if e.status_code == 404:
+        _LOGGER.error("{} bucket does not exist!".format(bucket_name))
+      else:
+        _LOGGER.warning(
+            "HttpError - unable to verify whether bucket {} exists.\
+        ".format(bucket_name))
```

Comment:
Please do not pre-format messages in logging statements. This affects the performance as the formatting can be ignored by the logger configuration while pre-formatting is always executed and may waste resources.

Instead, pass them as arguments:

```
_LOGGER.level('template %s message', arg1, arg2, arg3, ...)
```

Reply:
Noted, this should be resolved. Thanks!